### PR TITLE
Fix empty service account check

### PR DESCRIPTION
**Pull request by bnshr**

### CODE REVIEW

**Review by rdavid — APPROVED** [review]

**Comment by bnshr** [comment]
@aabughosh is notified to make the required changes for adding service account in the CRD deployment.

File: cnf-certification-test/accesscontrol/suite.go [hunk 1]
```diff
@@ -438,7 +439,7 @@ func testPodServiceAccount(env *provider.TestEnvironment) {
 	failedPods := []string{}
 	for _, put := range env.Pods {
 		ginkgo.By(fmt.Sprintf("Testing service account for pod %s (ns: %s)", put.Name, put.Namespace))
-		if put.Spec.ServiceAccountName == "" {
+		if put.Spec.ServiceAccountName == defaultServiceAccount {
```
Comment (edcdavid) [inline cnf-certification-test/accesscontrol/suite.go:442]: update catalog test description with detailed behavior (e.g. that we should not use the default service account)

Comment (bnshr) [reply cnf-certification-test/accesscontrol/suite.go:442]: Updated. Please review again!

**Review by sebrandon1 — APPROVED** [review]

File: cnf-certification-test/accesscontrol/suite.go [hunk 2]
```diff
@@ -553,7 +554,7 @@ func testAutomountServiceToken(env *provider.TestEnvironment) {
 	failedPods := []string{}
 	for _, put := range env.Pods {
 		ginkgo.By(fmt.Sprintf("check the existence of pod service account %s (ns= %s )", put.Namespace, put.Name))
-		if put.Spec.ServiceAccountName == "" {
+		if put.Spec.ServiceAccountName == defaultServiceAccount {
 			tnf.ClaimFilePrintf("Pod %s has been found with an empty service account name.", put.Name)
```
Comment (sebrandon1) [inline cnf-certification-test/accesscontrol/suite.go:558]: This should probably be changed to:

```suggestion
			tnf.ClaimFilePrintf("Pod %s has been found with a default service account", put.Name)
```

**Review by sebrandon1 — COMMENTED** [review]
Just a quick note about the error message returned for the automount test.

**Comment by sebrandon1** [comment]
The tests seemed to be failing for the test-0, test-1, etc pods in the -partner repo so I had to an explicit service account name there: https://github.com/test-network-function/cnf-certification-test-partner/pull/299

I also added a note in the cnf-best-practices doc: https://github.com/test-network-function/cnf-best-practices/pull/13

File: cnf-certification-test/accesscontrol/suite.go (hunk 2 above)
Comment (bnshr) [reply cnf-certification-test/accesscontrol/suite.go]: Updated. Thanks!

**Pull request merged**